tests/rbd_mirror/test_cg_mirror.py

```diff
@@ -0,0 +1,397 @@
+    client_primary,
+    client_secondary,
+    primary_cluster,
+    secondary_cluster,
+    pool_types,
+    **kw
+):
+    """
+    Test user can successfully mirror rbd group containing rbd images between primary and secondary sites
+    Args:
+        rbd_obj: RBD object
+        client : client node object
+        **kw: any other arguments
+    """
+
+    for pool_type in pool_types:
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = deepcopy(getdict(rbd_config))
+
+        for pool, pool_config in multi_pool_config.items():
+            group_ls_kw = {}
+            if "data_pool" in pool_config.keys():
+                _ = pool_config.pop("data_pool")
+            group_ls_kw.update({"pool": pool})
+
+            for image, image_config in pool_config.items():
+                if "image" in image:
+                    # Add data to the images
+                    err = run_IO(rbd_primary, client_primary, pool, image, **kw)
+                    if err:
+                        return 1
+
+            # Get Groups in a Pool
+            (group, _) = rbd_primary.group.list(**group_ls_kw)
+            log.info("Groups present in pool " + pool + " are: " + group)
+            if _:
+                log.error("Listing Group in a pool Failed")
+                return 1
+            group_ls_kw.update({"group": group.strip()})
+
+            # Get Group Mirroring Status
+            (group_mirror_status, _) = rbd_primary.mirror.group.status(**group_ls_kw)
+            if _:
+                if "mirroring not enabled on the group" in _:
+                    mirror_state = "Disabled"
+                else:
+                    log.error("Getting group mirror status failed : " + str(_))
+                    return 1
+            else:
+                mirror_state = "Enabled"
+            log.info("Group " + group + " mirroring state is " + mirror_state)
+
+            # Enable Group Mirroring and Verify
+            if mirror_state == "Disabled":
+                out = enable_group_mirroring_and_verify_state(
+                    rbd_primary, **group_ls_kw
+                )
+                if out:
+                    log.error("Enabling group mirroring failed")
+                    return 1
+
+            # Wait for group mirroring to complete
+            out = wait_for_idle(rbd_primary, **group_ls_kw)
+            if out:
+                log.error("Group Mirorring is not idle after 300 seconds")
+                return 1
+
+            # Validate size of each image should be same on site-a and site-b
+            (group_image_list, _) = rbd_primary.group.image.list(
+                **group_ls_kw, format="json"
+            )
+            match = compare_image_size_primary_secondary(
+                rbd_primary, rbd_secondary, group_image_list
+            )
+            if not match:
+                log.error(
+                    "size of rbd images do not match on primary and secondary site"
```

Comment:
sometimes `rbd du` is not good option to check used size,
please explore on it with fio